### PR TITLE
Added some changes to the cmake file

### PR DESCRIPTION
**Pull request by cnegre**

### CODE REVIEW

**Comment by jeanlucf22** [comment]
@cnegre I don't understand why you need those changes. Which hardware/compiler are you building with? What errors do you get without these changes?

**Comment by cnegre** [comment]
The my code was not compiling unless I added bml libs on
target_link_libraries. This build system doesn't compile properly when we
use extra_cflags in bml.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,59 @@
...
 if(PROGRESS)
   enable_language(C CXX)
 endif()
 
+if(PROGRESS)
+  include(FindPkgConfig)
+  #Looks for the installation of progress in CMAKE_MODULE_PATH or CMAKE_PREFIX_PATH or whatever else cmake needs!
+  find_package(PROGRESS CONFIG QUIET)
+  #Checks if the PROGRESS modules are indeed available
+  pkg_check_modules(PROGRESS REQUIRED progress)
```
Comment (junghans) [inline CMakeLists.txt:19]: This is duplicated, either uses `pkg_check_modules()` or `find_package(... CONFIG)`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,26 +1,59 @@
...
+  list(APPEND BML_LIBRARIES ${BML_LDFLAGS})
+  message(STATUS "Found bml: ${BML_LDFLAGS}")
+else()
+  find_package(BLAS REQUIRED)
+  find_package(LAPACK REQUIRED)
+  message(STATUS "Linking BLAS via ${BLAS_LIBRARIES}")
+  message(STATUS "Linking LAPACK via ${LAPACK_LIBRARIES}")
+endif()
+
+list(APPEND LINK_LIBRARIES "-fopenmp -lm -Wl,--copy-dt-needed-entries")
```
Comment (junghans) [inline CMakeLists.txt:38]: And this feel like a hack, too!

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,26 +1,59 @@
...
+  #Checks if the PROGRESS modules are indeed available
+  pkg_check_modules(PROGRESS REQUIRED progress)
+  message(STATUS "Found progress: ${PROGRESS_LDFLAGS}")
+  #Appending the necesary flags into LINK_LIBRARIES
+  list(APPEND LINK_LIBRARIES ${PROGRESS_LDFLAGS})
+  list(APPEND PROGRESS_LIBRARIES ${PROGRESS_LDFLAGS})
+
+  #Same thing for BML!
+  find_package(BML CONFIG QUIET)
+  pkg_check_modules(BML REQUIRED bml)
```
Comment (junghans) [inline CMakeLists.txt:27]: Same here, pkg-config or find_package.

**Comment by junghans** [comment]
> The my code was not compiling unless I added bml libs on target_link_libraries. This build system doesn't compile properly when we use extra_cflags in bml.

The right way to handle extra cflags in CMake these days is to add something like:
```
target_compile_definitions(bml PUBLIC <the extra flags>)
```
in bml and then linking against `BML::bml` will automatically inherit all `PUBLIC` flags.

**Comment by jeanlucf22** [comment]
> The my code was not compiling unless I added bml libs on target_link_libraries. This build system doesn't compile properly when we use extra_cflags in bml.

What flag? What platform?

**Comment by cnegre** [comment]
@jeanlucf22  This is an old PR. Now I'm using your PR that was merged.  

Anyways:

Platform: Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz

Extra flags in progress to get threading:

export EXTRA_FCFLAGS=${EXTRA_FCFLAGS:=" -fopenmp -lpthread "}

**Comment by junghans** [comment]
> @jeanlucf22 This is an old PR. Now I'm using your PR that was merged.
> 
> Anyways:
> 
> Platform: Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
> 
> Extra flags in progress to get threading:
> 
> export EXTRA_FCFLAGS=${EXTRA_FCFLAGS:=" -fopenmp -lpthread "}

This seems hackish again, CMake should just call `find_package(OpenMP)` and then use `target_link_libraries(bml PUBLIC OpenMP::OpenMP_Fortran)`

I know when @nicolasbock wrote BML's CMake back when that wasn't an option, but we should just use a newer CMake Version now.

**Comment by nicolasbock** [comment]
I agree, it would be a good idea to update the `CMakeLists.txt` everywhere.

**Pull request closed without merging**